### PR TITLE
[RISCV] Support instruction sizes up to 176-bits in disassembler.

### PR DESCRIPTION
**Pull request by topperc**

### CODE REVIEW

**Comment by wangpc-pp** [comment]
Are we going to use larger instruction length?

**Comment by topperc** [comment]
> Are we going to use larger instruction length?

Not that I know of, but when I updated llvm-objdump in #90093 there was concern that I didn't handle 48-bit instructions. So I figured if that was really a concern we need to fix the disassembler too.

File: llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp [hunk 1]
```diff
@@ -656,12 +656,44 @@ DecodeStatus RISCVDisassembler::getInstruction(MCInst &MI, uint64_t &Size,
                                                ArrayRef<uint8_t> Bytes,
                                                uint64_t Address,
                                                raw_ostream &CS) const {
-  // TODO: This will need modification when supporting instruction set
-  // extensions with instructions > 32-bits (up to 176 bits wide).
+  // It's a 16 bit instruction if bit 0 and 1 are not 0x3.
+  if ((Bytes[0] & 0x3) != 0x3)
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp:660]: We can use `0b111`, `0b111111`, ..., which should be more readable.

File: llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp [hunk 2]
```diff
@@ -656,12 +656,44 @@ DecodeStatus RISCVDisassembler::getInstruction(MCInst &MI, uint64_t &Size,
                                                ArrayRef<uint8_t> Bytes,
                                                uint64_t Address,
                                                raw_ostream &CS) const {
-  // TODO: This will need modification when supporting instruction set
-  // extensions with instructions > 32-bits (up to 176 bits wide).
+  // It's a 16 bit instruction if bit 0 and 1 are not 0x3.
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp:659]:
```suggestion
  // It's a 16 bit instruction if bit 0 and 1 are not 0b11.
```

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Pull request merged**